src/main/java/edu/indiana/dlib/amppd/service/impl/BatchServiceImpl.java

```diff
@@ -389,27 +389,19 @@ private Primaryfile getPrimaryfile(Collection batchfileCollection, Item item, Ba
 		Primaryfile primaryFile =null;
 		boolean found = false;
 		Set <Primaryfile> primaryFiles = item.getPrimaryfiles();
+		
+		
+		// For primary files in a given item, file names must be unique.
 		if(primaryFiles != null && primaryFiles.size() >= 0) 
 		{
-			log.info("BATCH PROCESSING : loop to see if primary file name already exists for this item");
 			for(Primaryfile p : primaryFiles) 
 			{ 
 				if((p.getName() != null && p.getName().contentEquals(batchFile.getPrimaryfileName()) ) ) 
 				{
-					boolean matchesExternalId = item.getExternalId() != null && item.getExternalId().equals(batchFile.getExternalItemId());
```

Comment:
@yingfeng-iu I took this validation out because at this point we have determined found the Item that are within a collection/unit.  No need to validate ExternalSource here as it's was done when an item was selected/created.

Reply:
See line 436 of this file